library/Ivoz/Provider/Domain/Model/UsersCdr/UsersCdrDto.php

```diff
@@ -17,10 +17,28 @@ public static function getPropertyMap(string $context = '', string $role = null)
                 'disposition' => 'disposition',
                 'id' => 'id'
             ];
+
+            if ($role !== 'ROLE_COMPANY_USER') {
+                $response += [
+                    'userId' => 'user',
+                    'friendId' => 'friend',
+                    'extensionId' => 'extension',
+                ];
+            }
         } else {
             $response = parent::getPropertyMap(...func_get_args());
         }
 
+        if ($role === 'ROLE_BRAND_ADMIN') {
+            unset($response['brandId']);
+        } elseif ($role === 'ROLE_COMPANY_ADMIN') {
+            unset($response['brandId']);
+            unset($response['companyId']);
+        } elseif ($role === 'ROLE_COMPANY_USER') {
+            unset($response['brandId']);
+            unset($response['companyId']);
+        }
```

Comment:
This only applies to non-collection contexts of user API. There is no detailed view of UsersCdrs in that API right now, but it will be required sooner than later, so this is also debatable